### DISCUSSION TITLE
Wallet integration

### DISCUSSION THREAD

**Discussion by alexaunter** [galactic3/ft-lockup-contract#8] [Q&A] [answered]
In old implementation it seems like one account belongs to exactly one lockup account
(since factory generates lockup based on first 20 bytes of sha256 of account name, see https://github.com/near/core-contracts/blob/5f4b7638d4f446eeb089e261dc80c4dcaf69dd48/lockup-factory/src/lib.rs#L120
This gives us an ability to deterministically find lockup account name with no requests to the blockchain.

But in new implementation
1. single lockup can have multiple users
2. single user may need different lockups for different tokens

Was this problem already thought through, or do we need to come up with a solution to this problem?

**Comment by frol** [comment, marked as answer]
We discussed the opportunity to use Indexer facilities in order to find all the lockup contracts that user has. It should be fine going that route.

**Comment by alexaunter** [comment]
Quoting @frol:
NEAR team will implement lockup wallet integration themselves
We only work on the lockup contract + frontend and provide POC  for wallet integration.

**Reply by frol** [reply]
Well, some thin TS library that defines the types and public methods would be really handy for integrations.
